src/webgl/texture.js

```diff
@@ -210,6 +221,25 @@ export default class Texture extends Resource {
     };
     const glSettings = Object.assign({}, DEFAULT_TEXTURE_SETTINGS, pixelStore);
 
+    // NOTE: NPOT workaround: Force update default settings.
+    if (this._isNPOTWorkaroundApplicable()) {
+
+      log.warn(0, 'NPOT (non power of two) texture found, disabling mipmaping');
+      mipmaps = false;
+      if (parameters[this.gl.TEXTURE_MIN_FILTER] === undefined) {
```

Comment:
The `initialize()` function is a bit long, we should define helper functions and break it down.